### ISSUE TITLE
Any plan to include the app on Fdroid repo?

### ISSUE THREAD

**Issue by much-doge** [dreautall/waterfly-iii#2]
I would say this is a Firefly III android client which I like the most in terms of UI and ease of access, it would be nice for it to be included on Fdroid repository and I'm sure that way more people would know about it.

**Comment by much-doge** [comment]
awesome 👍

**Comment by natrius** [comment]
I have it.through the izzysoft repo, but it seems.to be stuck at 0.1.7-beta ?

**Comment by IzzySoft** [comment]
> I'm honestly not sure how Izzysoft determines if there is an app update.

```yaml
AutoUpdateMode: Version v%v
UpdateCheckMode: Tags
```

So my updater looks at the tags if there's a tag name indicating a higher version. Just triggered it manually now and it fetched v0.1.12-beta. Looking at the tag: That APK should have been fetched yesterday (and the other ones before that) – I have no idea why it didn't (the auto-update doesn't do anything different than the manually triggered one, just that the latter only considers the specified app). The logs show no errors either. Hope this was just a one-time hick-up; should it happen again, please give me a ping.

**Comment by kaanelloed** [comment]
@IzzySoft The app is still stuck at 0.1.7-beta , even with the manual update.

**Comment by IzzySoft** [comment]
@kaanelloed nothing I can do about that. 0.1.7-beta has versionCode 2096, so to Android that is newer than the 1993 from the current latest release:

    package: name='com.dreautall.waterflyiii' versionCode='1993' versionName='0.3.0-beta'

versionCode must never be decreased, or things like this happen. Even if I'd pull this latest one to my repo and remove the 0.1.7, it wouldn't be offered to you as update – and if you tried to update manually, Android would refuse telling you that downgrades are not allowed. So you'll have to wait until the "base versionCode" (aka `variant.versionCode`) reaches 210 (which then results in 2103 for arm64) before updates will show up in my repo.

If you wonder how this might have happened, I have a good guess. Most likely, initially Flutter's own versionCode generation was used. Flutter takes the "base versionCode" and adds `ABI * 1000` to it, with 2 for arm64. So with 0.1.7, the base versionCode probably was 96, which +2000 resulted in 2096. This calculation is a bit problematic, as it would count an older arm64 as newer than a recent armeabi. So the project decided to switch to a proper versioning, using the calculation `variant.versionCode *10 + ABI`. By that calculation, 0.1.7 would have had 963 (as arm64 [was re-defined to 3](https://github.com/dreautall/waterfly-iii/blob/master/android/app/build.gradle#L101) and 0.1.7 had 96 as base versionCode).

Maybe @dreautall wants to initialize the "jump" to `variant.versionCode = 210` to fix this gap?

**Comment by IzzySoft** [comment]
Yeah, thought the jump wouldn't be "too far" and thus acceptable :stuck_out_tongue_winking_eye: Of course I could have simply replaced the APK here, but that would leave those at 0.1.7 and below stranded for another while. So going directly to 210 with the next release would solve it for all.

Thanks a lot, @dreautall!

**Comment by IzzySoft** [comment]
Yupp, that worked out fine! Just the per-release changelog was not pulled as it could not be matched. So let me guess… I pick the arm64 one.  According to your `build.gradle` that's a `3`. At the very same it says `versionCode x 10 + abiCode`. I don't see where the base is established, but that would mean my 2133 must be `213.txt` – and I can calculate that by `(versionCode - 3) / 10`. OK, I've set that, next release will show if it works (added the current changelog manually for now).

Let me check some more while we're at Fastlane. My notes say I don't pick shortdesc because it's too long (limit is 80 chars). That seems to be mostly fixed, except for

* es-ES: 84
* fr-FR: 98
* pt-BR: 81
* ro-RO: 89

Could you please fix that?

fulldesc is also skipped due to formatting issues: it's close to Markdown, but would only render as Github flavored Markdown, which my renderer does not support. Could you make sure it renders as "standard Markdown"? That means, paragraphs separated by empty lines, and bullet-point lists too.

**Comment by IzzySoft** [comment]
> will update the files

Please let me know when it's done, so I can enable those here.

> Can I/you somehow mark it as plain text?

I can, but it won't look as good then (e.g. the longer lines in the bullet-point list would wrap on phone screens, thus destroying the formatting). All that would be needed to make it render as proper Markdown are two empty lines inserted: one before the bullet-point list, and one before the last line (after "Contribution:") – though the latter would not matter too much if you're fine with the two lines combined to a single paragraph.

**Comment by dreautall** [comment]
Hi, thank you for the hint! So same limitations then than the Play Store. I already shortened the description for the Play Store anyways (as I had to manually fill it in), I'll copy them over in the text files as well! Unfortunately I saw the text limit setting in Crowdin too late :(
